projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitPdfActivity.java

```diff
@@ -32,179 +39,243 @@
+    lawsuit =
+        lawsuit.replace(
+            "<companyId>", ((EditText) findViewById(R.id.companyId)).getText().toString());
+    lawsuit =
+        lawsuit.replace(
+            "<companyAddress>",
+            ((EditText) findViewById(R.id.companyAddress)).getText().toString());
+    lawsuit =
+        lawsuit.replace(
+            "<companyPhone>", ((EditText) findViewById(R.id.companyPhone)).getText().toString());
+    lawsuit =
+        lawsuit.replace(
+            "<companyFax>", ((EditText) findViewById(R.id.companyFax)).getText().toString());
 
-    // Output
-    lawsuitOutputPath = Paths.get(lawsuitMainPath, LAWSUIT_OUTPUT_DIR_NAME).toString();
-    makeDir(lawsuitOutputPath);
+    return lawsuit;
   }
 
-  private void makeDir(String path) {
-    File directory = new File(path);
-    if (directory.exists() && directory.isDirectory()) directory.mkdirs();
+  // Pops a dialog asking user if to share pdf
+  private void sharePdf(String absFilename) {
+    if (new File(absFilename).exists()) {
+      new AlertDialog.Builder(this, R.style.AlertDialog)
+          .setTitle("כתב תביעה נוצר")
+          .setMessage("כתב התביעה נוצר ונשמר במכשירך.\n האם תרצה/י לשתף?")
+          .setPositiveButton(
+              "כן",
+              (dialog, which) -> {
+                Uri uri =
+                    FileProvider.getUriForFile(
+                        getApplicationContext(),
+                        getPackageName() + ".fileprovider",
+                        new File(absFilename));
+                Intent shareIntent = new Intent();
+                shareIntent.setAction(Intent.ACTION_SEND);
+                shareIntent.setType("application/pdf");
+                shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
+                startActivity(Intent.createChooser(shareIntent, "Share via"));
+              })
+          .setNegativeButton("לא", (dialog, which) -> dialog.dismiss())
+          .create()
+          .show();
+    }
+    Log.w(TAG, "Could not share pdf, file doesn't exists.\nabsFilename: " + absFilename);
   }
 
-  private void setTimeZones() {
-    DATE_TIME_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
-    DATE_FORMATTER.setTimeZone(TimeZone.getTimeZone(TIME_ZONE));
+  // Pops a calendar dialog when clicking on date fields
+  private void displayDatePicker() {
+    calendar = Calendar.getInstance();
+    int day = calendar.get(Calendar.DAY_OF_MONTH);
+    int month = calendar.get(Calendar.MONTH);
+    int year = calendar.get(Calendar.YEAR);
+
+    datePickerDialog =
+        new /**/ DatePickerDialog(
+            LawsuitPdfActivity.this,
+            (datePicker, currentYear, currentMonth, currentDay) ->
+                receivedDate.setText(day + "/" + (month + 1) + "/" + year),
+            year,
+            month,
+            day);
+    datePickerDialog.show();
   }
 
-  private void checkPermissionsThenCreatePdf() {
+  // TODO: Move checkPermissions(), onRequestPermissionsResult(), createOutputDir() to
```

Comment:
We decided to leave the permission request here and not moving it to the splash screen.

Reply:
Removed it...